rclpy/test/test_lifecycle.py

```diff
@@ -0,0 +1,119 @@
+    node.destroy_node()
+    node = LifecycleNode('test_lifecycle_node_init_2', enable_communication_interface=False)
+    assert node
+    # Make sure services were not created
+    assert not hasattr(node, '_service_change_state')
+    assert not hasattr(node, '_service_get_state')
+    assert not hasattr(node, '_service_get_available_states')
+    assert not hasattr(node, '_service_get_available_transitions')
+    assert not hasattr(node, '_service_get_transition_graph')
+    # Make sure also that the services were not created in the pybind11 plugin
+    assert not node._state_machine.service_change_state
+    assert not node._state_machine.service_get_state
+    assert not node._state_machine.service_get_available_states
+    assert not node._state_machine.service_get_available_transitions
+    assert not node._state_machine.service_get_transition_graph
+    node.destroy_node()
+    with pytest.raises(TypeError):
+        LifecycleNode('test_lifecycle_node_init_3', enable_communication_interface='asd')
+
+
+def test_lifecycle_state_transitions():
+    node = LifecycleNode(
+        'test_lifecycle_state_transitions_1', enable_communication_interface=False)
+    # normal transitions
+    assert node.trigger_configure() == TransitionCallbackReturn.SUCCESS
+    assert node.trigger_activate() == TransitionCallbackReturn.SUCCESS
+    assert node.trigger_deactivate() == TransitionCallbackReturn.SUCCESS
+    assert node.trigger_cleanup() == TransitionCallbackReturn.SUCCESS
+    # some that are not possible from the current state
+    with pytest.raises(_rclpy.RCLError):
+        node.trigger_activate()
+    with pytest.raises(_rclpy.RCLError):
+        node.trigger_deactivate()
+    assert node.trigger_shutdown() == TransitionCallbackReturn.SUCCESS
+    node.destroy_node()
+
+    class ErrorOnConfigureHandledCorrectlyNode(LifecycleNode):
+
+        def on_configure(self):
+            return TransitionCallbackReturn.ERROR
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+    node = ErrorOnConfigureHandledCorrectlyNode(
+        'test_lifecycle_state_transitions_2', enable_communication_interface=False)
+    assert node.trigger_configure() == TransitionCallbackReturn.ERROR
+    assert node._state_machine.current_state[1] == 'unconfigured'
+
+    class ErrorOnConfigureHandledInCorrectlyNode(LifecycleNode):
+
+        def on_configure(self):
+            return TransitionCallbackReturn.ERROR
+
+        def on_error(self):
+            return TransitionCallbackReturn.ERROR
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+    node = ErrorOnConfigureHandledInCorrectlyNode(
+        'test_lifecycle_state_transitions_3', enable_communication_interface=False)
+    assert node.trigger_configure() == TransitionCallbackReturn.ERROR
+    assert node._state_machine.current_state[1] == 'finalized'
+
+# TODO(ivanpauno): Add automated tests for lifecycle services!!
```

Comment:
Do you plan to do this in this PR?

Reply:
Done in https://github.com/ros2/rclpy/pull/865/commits/928ee797ce0c72cb0b3aa80f3d5de797e5d6da6c.